scientific_library/tvb/simulator/simulator.py

```diff
@@ -253,29 +277,38 @@ def configure(self, full_configure=True):
             The configured Simulator instance.
 
         """
+        # TODO: temporary hack because numba dfuns fail for multiple modes
+        if self.use_numba and self.model.number_of_modes == 1:
```

Comment:
Shouldn't we better fix those numba dfuns, and not commit this `temporary hack` ?

Reply:
It is not that easy for me at least. I don't understand 100% how the numba arrays work.

But, even if we fix this, we cannot expect that all users who want to add models, or have their own models sitting somewhere, they know how to write numba code.

I think that for some time, we should keep the option for the user to choose the numpy or the numba funs, somehow.

Setting the spatial_reshape shape correctly would follow then, depending on whether we have written the numba funs for multiple modes or not.

Still, I don't know if this solution of having a use_numba flag is the best way to deal with this...

Reply:
I think there is already a mecanism to Avoid numba, by not inheriting numba based model class, and not Having a numba function.

Reply:
As for avoiding numba, I meant at runtime.